### PR TITLE
Deny connection to websocket coming from different origin in Nodejs server

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Validates each request with a session token
- Checks for origin for websocket upgrade request and denies connection if from a different origin

Build: https://builds.cask.co/browse/CDAP-UDUT386

### CODE REVIEW

File: cdap-ui/server.js [hunk 1]
```diff
@@ -39,77 +39,108 @@ var cdapConfig, securityConfig;
...
-        delete a;
+      c.on('close', function() {
+        a = null;
       });
     });
 
     sockServer.installHandlers(server, { prefix: '/_sock' });
+
+    server.addListener('upgrade', function(req, socket) {
+      const nodejsserver = cdapConfig['dashboard.bind.address'];
```
Comment (itsanudeep) [inline cdap-ui/server.js:119]: nit: lowerCamelCase for nodejsserver, nodejsport, nodejsprotocol?

**Review by itsanudeep — COMMENTED** [review]
nit, otherwise LGTM :+1:

**Comment by elfenheart** [comment]
I couldn't navigate to our angular apps

**Comment by chtyim** [comment]
The token generation and verification part LGTM

**Review by elfenheart — APPROVED** [review]
@itsanudeep comment, otherwise 👍 LGTM

**Comment by ajainarayanan** [comment]
@itsanudeep @elfenheart PR rebased and updated. Could you give it another review?

**Comment by ajainarayanan** [comment]
Latest green build: https://builds.cask.co/browse/CDAP-UDUT386-5

**Pull request merged**